### PR TITLE
ci: re-run failed test steps with debug logging enabled

### PR DESCRIPTION
**Pull request by JKRhb**
This PR introduces a retry mechanism with enabled debug logging for failed CI testing steps as discussed in https://github.com/eclipse-thingweb/node-wot/pull/1184#issuecomment-1845510736.

### CODE REVIEW

**Comment by JKRhb** [comment]
Enabling debug logging on a second try does seem to work :) See this run for example: https://github.com/eclipse-thingweb/node-wot/actions/runs/7165425102/job/19507350139?pr=1189

**Review by danielpeintner — APPROVED** [review]
That may become very handy!

**Comment by JKRhb** [comment]
> Should we go for `test:only` the second time? or do we want always the coverage? even for debugging?

Hmm, I would assume that if the retry should succeed using the `test:only` script. this would cause the codecov step to fail instead 🤔 However, always failing the retry eventually to signal that something went wrong might be the right approach here anyway, I suppose?

**Comment by relu91** [comment]
> However, always failing the retry eventually to signal that something went wrong might be the right approach here anyway, I suppose?

Yes, I think we should always fail if we re-try... but maybe not because the coverage step should fail... we should signal it better. Is it possible to configure the action to fail if it re-tries?

**Comment by JKRhb** [comment]
> > However, always failing the retry eventually to signal that something went wrong might be the right approach here anyway, I suppose?
> 
> Yes, I think we should always fail if we re-try... but maybe not because the coverage step should fail... we should signal it better. Is it possible to configure the action to fail if it re-tries?

This should now be incorporated :) "Unfortunately", the MQTT tests have not been failing anymore so the behavior in this specific case could not really be tested by me so far. However, after triggering a "manual" the updated action step (more or less) showed the correct behavior.

**Comment by JKRhb** [comment]
Hmm, as the failed CI run shows, the current approach does not work for Windows yet and needs to be slightly adjusted...

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -48,8 +48,16 @@ jobs:
...
+              uses: nick-fields/retry@v2
+              with:
+                  timeout_minutes: 10
+                  max_attempts: 2
+                  command: npm run coverage:only
+                  # Rerun with debug logging enabled on error, but always fail the retry
+                  new_command_on_retry: |
+                      DEBUG=node-wot* npm run test:only
+                      exit 1
+                  shell: bash
```
[the remark this answers is not in the document]
Comment (relu91) [reply .github/workflows/ci.yaml:60]: I think it is reasonable. @danielpeintner ?

[the remark this answers is not in the document]
Comment (danielpeintner) [reply .github/workflows/ci.yaml:60]: Fine by me. If we ever run into issues we can revert the change.
PR is good to be merge from my side 👍

**Pull request merged**